spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalTaskLauncher.java

```diff
@@ -111,6 +118,26 @@ public String launch(AppDeploymentRequest request) {
 		return taskLaunchId;
 	}
 
+	private void pruneTaskInstanceHistory(String taskDefinitionName, String taskLaunchId) {
```

Comment:
The only issue I see is that as we would be returning false `unknown` status for tasks that have been purged.   i.e. if I hit the /runtime/apps/id I will get `unknown`s for tasks that have successfully completed(then purged).  And we will have at most only one `complete` entry at a time for tasks.  Maybe this is a @sabbyanandan question.   Do people use this to check the status of running tasks?   Or should they be referred to the /tasks/executions endpoint?

Reply:
We have had reports of status being inconsistent in the list, summary, and executions pages. So, yeah, we have users using the different options. However, the "runtime" tab would show what is running, though. Do we show the tasks in that tab even if they are destroyed?

Reply:
Just took a look at the controller, only the streamDeployer is used to return statuses from the runtime/apps.   So I guess the question is, "Does anyone really use this feature in the local deployer?"